src/main/java/de/dennisguse/opentracks/TrackStoppedActivity.java

```diff
@@ -127,66 +127,74 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     private void storeTrackMetaData(ContentProviderUtils contentProviderUtils, Track track) {
-        Map<String, Object> run = new HashMap<>();
-        TrackStatistics trackStatistics = track.getTrackStatistics();
-        if (track.getId() != null) {
-            run.put("id", track.getId().id());
-        }
-
-        run.put("ascent",  trackStatistics.hasAltitudeMax() ? trackStatistics.getMaxAltitude() : 0); // if didnt go up +Infinity
-        run.put("descent", trackStatistics.hasAltitudeMin()? trackStatistics.getMinAltitude(): 0); // if didn't go down -Infinity
-        run.put("distance", trackStatistics.getTotalDistance().toM()); // meters
-        run.put("maxSpeed", trackStatistics.getMaxSpeed().toMPS());
-        run.put("movingTime", trackStatistics.getMovingTime().toMillis()); // ms
-        run.put("stoppedTime", trackStatistics.getStopTime().toEpochMilli());
-        run.put("timerTime", trackStatistics.getMovingTime().toMillis()); //ms
-        run.put("user", "TerrylAndAxel"); // TODO: get current user?
-
-        FirestoreCRUDUtil firestoreCRUD = FirestoreCRUDUtil.getInstance();
-        firestoreCRUD.createEntry(CRUDConstants.RUNS_TABLE, run);
 
+        final String trackId = "track_01";
 
-        //here are the commented out tests for Create and Read of user
-        //Map<String, Object> user = new HashMap<>();
-        //user.put("firstname", "test2user");
-        //user.put("lastname", "test2lastnameuser");
-        //user.put("id", "2");
-        //firestoreCRUD.createUser(user);
+        //CREATE
+        ActionCallback createCallback = new ActionCallback() {
+            @Override
+            public void onSuccess() {
+                Log.d(CRUDConstants.TAG_CREATED, "New version called!");
+            }
 
-        //firestoreCRUD.getUser("jNRwb1WmNMT4FXKh3hiq");
+            @Override
+            public void onFailure() {
+                //pass
+            }
+        };
 
+        FirestoreCRUDUtil.getInstance().createEntry(CRUDConstants.RUNS_TABLE, trackId, track.toJSON(), createCallback);
 
+        //UPDATE
+        ActionCallback updateCallback = new ActionCallback() {
+            @Override
+            public void onSuccess() {
+                Log.d(CRUDConstants.TAG_UPDATED, "New version called!");
```

Comment:
this applies to here as well, and next occurences

Reply:
see above